### ISSUE TITLE
"System.ArgumentNullException: Parameter "itemSpec" cannot be null." when building for Mac Catalyst on arm64 version of .NET

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#10845]
### Steps to Reproduce

1. Get an M1 machine.
2. Install .NET/ARM64: [dotnet-sdk-6.0.100-preview.2.21155.3-osx-arm64.pkg](https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-preview.2.21155.3/dotnet-sdk-6.0.100-preview.2.21155.3-osx-arm64.pkg) - interestingly the problem does not occur with the x64 version of .NET.
3. Install [Microsoft.MacCatalyst.Bundle.14.3.100-ci.main.337.pkg](https://bosstoragemirror.azureedge.net/wrench/main/98c8649d0c7d1e3c4c8d8d09e022befa853fb1e7/4541181/package/notarized/Microsoft.MacCatalyst.Bundle.14.3.100-ci.main.337.pkg)
4. Clone/checkout https://github.com/dotnet/net6-mobile-samples/commit/e052322d68c4da8a892e22eb776307102bf079be
5. cd HelloMacCatalyst && dotnet build

### Expected Behavior

Successful build

### Actual Behavior

```
$ dotnet build /bl
Microsoft (R) Build Engine version 16.10.0-preview-21126-01+6819f7ab0 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

/usr/local/share/dotnet/sdk/6.0.100-preview.2.21155.3/MSBuild.dll -consoleloggerparameters:Summary -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/usr/local/share/dotnet/sdk/6.0.100-preview.2.21155.3/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/usr/local/share/dotnet/sdk/6.0.100-preview.2.21155.3/dotnet.dll -maxcpucount -restore -verbosity:m /bl ./HelloMacCatalyst.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  Detected signing identity:
          
    Bundle Id: com.microsoft.net6.hellomaccatalyst
    App Id: com.microsoft.net6.hellomaccatalyst
  HelloMacCatalyst -> /Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/bin/Debug/net6.0-maccatalyst/maccatalyst-x64/HelloMacCatalyst.dll
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error : System.ArgumentNullException: Parameter "itemSpec" cannot be null. [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Microsoft.Build.Shared.ErrorUtilities.VerifyThrowArgumentNull(Object parameter, String parameterName, String resourceName) [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Microsoft.Build.Utilities.TaskItem..ctor(String itemSpec) [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Xamarin.MacDev.Tasks.SmartCopyTaskBase.CopyFile(String source, String target, String targetItemSpec) in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/SmartCopyTaskBase.cs:line 74 [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Xamarin.MacDev.Tasks.SmartCopyTaskBase.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/SmartCopyTaskBase.cs:line 99 [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]

Build FAILED.

/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error : System.ArgumentNullException: Parameter "itemSpec" cannot be null. [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Microsoft.Build.Shared.ErrorUtilities.VerifyThrowArgumentNull(Object parameter, String parameterName, String resourceName) [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Microsoft.Build.Utilities.TaskItem..ctor(String itemSpec) [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Xamarin.MacDev.Tasks.SmartCopyTaskBase.CopyFile(String source, String target, String targetItemSpec) in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/SmartCopyTaskBase.cs:line 74 [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
/usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/14.3.100-ci.main.337/tools/msbuild/iOS/Xamarin.Shared.targets(605,3): error :    at Xamarin.MacDev.Tasks.SmartCopyTaskBase.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/SmartCopyTaskBase.cs:line 99 [/Users/rolf/work/net6-mobile-samples/HelloMacCatalyst/HelloMacCatalyst.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.54
```

### Build Logs

[msbuild.binlog.zip](https://github.com/xamarin/xamarin-macios/files/6125442/msbuild.binlog.zip)

**Comment by AlaaAlHallaq** [comment]
same with me.

**Comment by jsaunders92** [comment]
I'm also seeing this issue!

**Comment by u2m4c6** [comment]
Same thing here. Any updates from the team?

**Comment by rolfbjarne** [comment]
This was fixed here: 8bec55e22ad996566ac77338aa8559a33ef8616f

The fix will be included in Preview 6.
